src/slip.rs

```diff
@@ -0,0 +1,200 @@
+use secp256k1::{PublicKey};
+
+/// A UTXO slip containing record of owernship of funds on the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Slip {
```

Comment:
My understanding of UTXO is like this: For the sake of performance, an input/output(what we are calling "slip") must be spent entirely within a transaction.

With this understanding, a slip should have a field called ID or maybe input(e.g. blockid+slip index) which can identify the unspent output that's being spent. it should also have a list(vec or []) of "outputs" which are each address+amount. The work of finding which outputs are appropriate for a given transaction the user wants to do is put onto the wallet, the UTXOSet validation for the node is then just to check for the presence of the outputs in the UTXOSet by ID.

In total, the structures should look like this: A transaction has one sig and multiple slips, a slip can have multiple outputs.

Reply:
Spoke with Stephen and resolved my concerns here. I had a misunderstanding about the to/from fields on transaction.